electrum/gui/qt/main_window.py

```diff
@@ -1967,12 +1967,17 @@ def on_pr(self, request: 'paymentrequest.PaymentRequest'):
         else:
             self.payment_request_error_signal.emit()
 
-    def parse_lightning_invoice(self, invoice):
+    def set_ln_invoice(self, invoice: str):
         """Parse ln invoice, and prepare the send tab for it."""
         try:
             lnaddr = lndecode(invoice)
+        except LnInvoiceException as e:
+            self.show_error(_("Error parsing Lightning invoice") + f":\n{e}")
+            return
         except Exception as e:
-            raise LnDecodeException(e) from e
+            raise LnInvoiceException(e) from e
```

Comment:
Is the intention here to propagate this exception to the crash reporter?
Note that even without catching and re-raising it with a different type, i.e. without these two lines, this exception would get to the crash reporter anyway.

So what is the intention of these two lines; specifically re re-raising the original exception? One side-effect it will have is that the crash reporter might group the reports together (irrespective of the original exception) as it only looks at the last exception. ( Reports are grouped by `id`, see: https://github.com/spesmilo/electrum/blob/d0f0669e8f7519018b6c79958184dabf467cf4ff/electrum/base_crash_reporter.py#L85-L89 )

Reply:
Ah, ok then this doesn't make so much sense. I just repeated the pattern that was existent before. Removed the re-raising.